launcher/container_runner.go

```diff
@@ -609,34 +610,61 @@ func initImage(ctx context.Context, cdClient *containerd.Client, launchSpec spec
 	return image, nil
 }
 
-// openPorts writes firewall rules to accept all traffic into that port and protocol using iptables.
+// openPorts writes firewall rules to accept all traffic into that port and protocol using iptables and ip6tables.
 func openPorts(ports map[string]struct{}) error {
-	for k := range ports {
-		portAndProtocol := strings.Split(k, "/")
-		if len(portAndProtocol) != 2 {
-			return fmt.Errorf("failed to parse port and protocol: got %s, expected [port]/[protocol] 80/tcp", portAndProtocol)
+	tcpPorts, udpPorts, err := getPortsList(ports)
+	if err != nil {
+		return err
+	}
+
+	portMap := make(map[string]string)
+	portMap["tcp"] = tcpPorts
+	portMap["udp"] = udpPorts
+
+	for _, command := range []string{"iptables", "ip6tables"} {
+		for protocol, ports := range portMap {
+			if ports == "" {
+				continue
+			}
+			cmd := exec.Command(command, "-A", "INPUT", "-p", protocol, "-m", "multiport", "--dport", ports, "-j", "ACCEPT")
+			out, err := cmd.CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("failed to set firewall policy using %v, for %s %s: %v %s", cmd, ports, protocol, err, out)
```

Comment:
Do we want to collect all errors and then return, or return on error?

Reply:
I think here one error is fine, all four commands will probably result in the same error if there is one.